test/static.test.js

```diff
@@ -2283,3 +2283,78 @@ t.test('routes use default errorHandler when fastify.errorHandler is not defined
     })
   })
 })
+
+t.test('will serve pre-compressed files with .br at the highest priority', async (t) => {
+  const pluginOptions = {
+    root: path.join(__dirname, '/static-pre-compressed'),
+    prefix: '/static-pre-compressed/',
+    preCompressed: true
+  }
+
+  const fastify = Fastify()
+
+  fastify.register(fastifyStatic, pluginOptions)
+  t.tearDown(fastify.close.bind(fastify))
+
+  const response = await fastify.inject({
+    method: 'GET',
+    url: '/static-pre-compressed/all-three.html',
+    headers: {
+      'accept-encoding': 'gzip, deflate, br'
+    }
+  })
+
+  t.strictEqual(response.headers['content-encoding'], 'br')
+  t.strictEqual(response.statusCode, 200)
```

Comment:
Can you please check that if the precompressed body matches?

Reply:
Matches what exactly? The content on disk (similar to the other tests with the top level ` fs.readFileSync` calls)?

Reply:
Yes.